### PR TITLE
2.0.2 cherry pick

### PR DESCRIPTION
**Pull request by erimatnor**
Cherry-picked commits for the 2.0.2 release.

### CODE REVIEW

**Comment by svenklemm** [comment]
needs https://github.com/timescale/timescaledb/pull/2957 for sanitizer test to pass

**Comment by svenklemm** [comment]
https://github.com/timescale/timescaledb/pull/2959

**Review by pmwkaa — APPROVED** [review]

**Review by mkindahl — APPROVED** [review]
LGTM

File: appveyor.yml [hunk 1]
```diff
@@ -191,7 +191,7 @@ test_script:
 
     #right now we only run timescale regression tests, others will be set up later
 
-    docker exec -e IGNORES="chunk_utils loader" -e TEST_TABLESPACE1_PREFIX="C:\Users\$env:UserName\Documents\tablespace1\" -e TEST_TABLESPACE2_PREFIX="C:\Users\$env:UserName\Documents\tablespace2\" -e TEST_SPINWAIT_ITERS=10000 -e USER=postgres -it pgregress /bin/bash -c "cd /timescaledb/build && make regresschecklocal"
```
Comment (svenklemm) [inline appveyor.yml:194]: needs to be adjusted cause the cluster test isn't version specific in 2.0.x

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,60 @@
...
+
+**Minor features**
+* #2736 Support adding columns to hypertables with compression enabled
+* #2926 Optimize cagg refresh for small invalidations
+
+**Bugfixes**
+* #2883 Fix join qual propagation for nested joins
+* #2908 Fix changing column type of clustered hypertables
+* #2942 Validate continuous aggregate policy
+
```
Comment (mkindahl) [inline CHANGELOG.md:17]: Shouldn't you list all issues that are cherry-picked? I'm missing issue #2831 and issue #2770 (or PR #2893)

File: scripts/test_sanitizers.sh [hunk 3]
```diff
@@ -101,5 +101,5 @@ echo "Testing"
 # postmaster.c is not atomic but read/written across signal handlers
 # and ServerLoop.
 docker exec -i -u postgres -w /tsdb_build/timescaledb/build timescaledb-san /bin/bash <<EOF
-make -k regresscheck regresscheck-t SKIPS='remote_txn' IGNORES='bgw_db_scheduler bgw_launcher continuous_aggs_ddl-11'
```
Comment (svenklemm) [inline scripts/test_sanitizers.sh:104]: needs to be adjusted cause the cluster test is not version specific in 2.0.x

**Review by svenklemm — APPROVED** [review]

**Pull request merged**